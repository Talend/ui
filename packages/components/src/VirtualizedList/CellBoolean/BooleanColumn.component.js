import React from 'react';
import { defaultColumnConfiguration } from '../Content.component';
import CellBoolean from './CellBoolean.component';

export const cellType = 'texticon';
export const booleanColumnConfiguration = {
	cellRenderer: props => <CellBoolean {...props} />,
};

// this is a fake component to be usable in JSX,
// but the element is used as props object internally (VirtualizedList / RV)
export default function BooleanColumn() {
	return null;
}

BooleanColumn.defaultProps = {
	...defaultColumnConfiguration,
	...booleanColumnConfiguration,
};
