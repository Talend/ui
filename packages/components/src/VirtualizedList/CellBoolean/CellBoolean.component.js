import PropTypes from 'prop-types';
import React from 'react';
import classnames from 'classnames';
import { withTranslation } from 'react-i18next';

import I18N_DOMAIN_COMPONENTS from '../../constants';
import getDefaultT from '../../translate';

import Action from '../../Actions/Action';
import styles from './CellBoolean.scss';

/**
 * Cell renderer that displays a boolean
 */
class CellBoolean extends React.Component {
	shouldComponentUpdate(nextProps) {
		return this.props.cellData !== nextProps.cellData;
	}

	render() {
		const { cellData, t } = this.props;

		return (
			<div className={classnames('cell-boolean-container', styles['cell-boolean-container'])}>
				{ cellData === true && t('BOOLEAN_VALUE_TRUE', { defaultValue: 'Yes' }) }
				{ cellData === false && t('BOOLEAN_VALUE_FALSE', { defaultValue: 'No' }) }
			</div>
		);
	}
}

CellBoolean.displayName = 'VirtualizedList(CellBoolean)';
CellBoolean.propTypes = {
	// The cell value : props.rowData[props.dataKey]
	cellData: PropTypes.string,
	t: PropTypes.func,
};

CellBoolean.defaultProps = {
	t: getDefaultT(),
};

export default withTranslation(I18N_DOMAIN_COMPONENTS)(CellBoolean);
