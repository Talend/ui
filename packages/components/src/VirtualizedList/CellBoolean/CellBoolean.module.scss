@use '~@talend/bootstrap-theme/src/theme/guidelines' as *;

.cell-boolean-container {
	display: flex;
	width: 100%;
	justify-content: center;
}
