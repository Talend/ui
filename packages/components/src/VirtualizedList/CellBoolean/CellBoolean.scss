.cell-icon-container {
	display: flex;
}
