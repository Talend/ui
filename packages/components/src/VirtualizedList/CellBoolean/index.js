import BooleanColumn, { cellType, booleanColumnConfiguration } from './BooleanColumn.component';

export { cellType, BooleanColumn };
export default booleanColumnConfiguration;
