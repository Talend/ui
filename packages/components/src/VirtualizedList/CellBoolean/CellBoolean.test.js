import React from 'react';
import { shallow } from 'enzyme';

import CellBoolean from './CellBoolean.component';

describe('CellBoolean', () => {
	it('should render an empty cell', () => {
		const wrapper = shallow(<CellBoolean />);

		expect(wrapper.getElement()).toMatchSnapshot();
	});

	it('should render a truthy value', () => {
		const wrapper = shallow(<CellBoolean cellData />);

		expect(wrapper.getElement()).toMatchSnapshot();
	});

	it('should render a falsy value', () => {
		const wrapper = shallow(<CellBoolean cellData={false} />);

		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
