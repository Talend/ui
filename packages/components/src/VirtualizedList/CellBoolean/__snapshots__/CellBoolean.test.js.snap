// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellBoolean should render a falsy value 1`] = `
<VirtualizedList(CellBoolean)
  cellData={false}
  i18n={
    Object {
      "changeLanguage": [Function],
      "getFixedT": [Function],
      "hasResourceBundle": [Function],
      "init": [Function],
      "isInitialized": true,
      "isMock": true,
      "language": "en",
      "languages": Array [
        "en",
      ],
      "loadNamespaces": [Function],
      "off": [Function],
      "on": [Function],
      "options": Object {},
      "reportNamespaces": ReportNamespaces {
        "usedNamespaces": Object {
          "tui-components": true,
        },
      },
      "services": Object {
        "backendConnector": Object {},
        "resourceStore": Object {
          "data": Object {},
        },
      },
      "store": Object {
        "data": Object {},
        "off": [Function],
        "on": [Function],
      },
      "t": [Function],
    }
  }
  t={[Function]}
  tReady={true}
/>
`;

exports[`CellBoolean should render a truthy value 1`] = `
<VirtualizedList(CellBoolean)
  cellData={true}
  i18n={
    Object {
      "changeLanguage": [Function],
      "getFixedT": [Function],
      "hasResourceBundle": [Function],
      "init": [Function],
      "isInitialized": true,
      "isMock": true,
      "language": "en",
      "languages": Array [
        "en",
      ],
      "loadNamespaces": [Function],
      "off": [Function],
      "on": [Function],
      "options": Object {},
      "reportNamespaces": ReportNamespaces {
        "usedNamespaces": Object {
          "tui-components": true,
        },
      },
      "services": Object {
        "backendConnector": Object {},
        "resourceStore": Object {
          "data": Object {},
        },
      },
      "store": Object {
        "data": Object {},
        "off": [Function],
        "on": [Function],
      },
      "t": [Function],
    }
  }
  t={[Function]}
  tReady={true}
/>
`;

exports[`CellBoolean should render an empty cell 1`] = `
<VirtualizedList(CellBoolean)
  i18n={
    Object {
      "changeLanguage": [Function],
      "getFixedT": [Function],
      "hasResourceBundle": [Function],
      "init": [Function],
      "isInitialized": true,
      "isMock": true,
      "language": "en",
      "languages": Array [
        "en",
      ],
      "loadNamespaces": [Function],
      "off": [Function],
      "on": [Function],
      "options": Object {},
      "reportNamespaces": ReportNamespaces {
        "usedNamespaces": Object {
          "tui-components": true,
        },
      },
      "services": Object {
        "backendConnector": Object {},
        "resourceStore": Object {
          "data": Object {},
        },
      },
      "store": Object {
        "data": Object {},
        "off": [Function],
        "on": [Function],
      },
      "t": [Function],
    }
  }
  t={[Function]}
  tReady={true}
/>
`;
